src/ipld/util.rs

```diff
@@ -363,18 +364,21 @@ pin_project! {
         queue: Vec<Cid>,
         fail_on_dead_links: bool,
     }
-}
 
-impl<DB, T> UnorderedChainStream<DB, T> {
-    pub fn into_seen(self) -> CidHashSet {
-        match Arc::try_unwrap(self.seen) {
-            Ok(v) => v.into_inner(),
-            Err(v) => v.lock().clone(),
+    impl<DB, T> PinnedDrop for UnorderedChainStream<DB, T> {
```

Comment:
I didn't find doc for implementing `PinnedDrop` with `pin-project-lite`, have you verified `drop` is being called?

Reply:
https://github.com/taiki-e/pin-project-lite/issues/61
https://github.com/taiki-e/pin-project-lite/pull/25

Reply:
Thanks!